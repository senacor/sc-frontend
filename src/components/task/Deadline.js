import React from 'react';
import Typography from 'material-ui/Typography';
import * as date from '../../helper/date';

const Deadline = props => {
  if (!props.deadline) {
    return null;
  }

  return (
    <Typography component="p">
      Deadline: {date.formatMomentForFrontend(props.deadline)}
    </Typography>
  );
};

export default Deadline;
