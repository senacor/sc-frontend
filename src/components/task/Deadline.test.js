import React from 'react';
import Deadline from './Deadline';
import { shallow } from 'enzyme';

it('displays the deadline if specified', () => {
  const deadline = '2017-11-31';

  const component = shallow(<Deadline deadline={deadline} />);

  expect(component).toMatchSnapshot();
});

it('displays nothing if no deadline is specified', () => {
  const component = shallow(<Deadline />);

  expect(component).toMatchSnapshot();
});
