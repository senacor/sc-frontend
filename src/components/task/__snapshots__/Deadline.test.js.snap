// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`displays nothing if no deadline is specified 1`] = `""`;

exports[`displays the deadline if specified 1`] = `
<WithStyles(Typography)
  component="p"
>
  Deadline: 
  Invalid date
</WithStyles(Typography)>
`;
